fern/docs/pages/getting-started/architecture.md

```
@@ -0,0 +1,250 @@
+<img src="../figures/data-flow/nixl_flow_01_init_light.svg" alt="Initialization sequence diagram showing agent creation, backend registration, and memory registration" style={{maxWidth: '78%'}} />
+</Frame>
+</div>
+<div className="diagram-dark">
+<Frame caption="Phase 1: Initialization">
+<img src="../figures/data-flow/nixl_flow_01_init_dark.svg" alt="Initialization sequence diagram showing agent creation, backend registration, and memory registration" style={{maxWidth: '78%'}} />
+</Frame>
+</div>
+
+The runtime creates a Transfer Agent per node and optionally passes a device list. It then calls `create_transfer_backend` for each desired transport. Memory segments are registered via `register_memory`, which creates the internal structures each backend needs. The application can target specific backends or let NIXL register with all backends that support the segment's memory type.
+
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)